examples/experimental/i3s-17-and-debug/tile-debug.js

```diff
@@ -144,9 +226,44 @@ function checkLOD(tile, tileWarnings) {
   tileWarnings.push({type: LOD_WARNING_TYPE, title});
 }
 
-function checkGeometryVsTexture(tile, tileWarnings) {
-  // TODO Geometry vs Texture Validation
-  return;
+function getTextureSize(tile) {
+  if (!tile.content) {
+    return 0;
+  }
+  const texture =
+    (tile.content.material &&
+      tile.content.material.pbrMetallicRoughness &&
+      tile.content.material.pbrMetallicRoughness.baseColorTexture &&
+      tile.content.material.pbrMetallicRoughness.baseColorTexture.texture.source.image) ||
+    tile.content.texture ||
+    null;
+  if (!texture) {
+    return 0;
+  }
+  return texture.height * texture.width;
+}
+
+function getTriangleVertices(attribute, offset) {
+  const geometryVertices = [];
+  for (let i = 0; i < 3; i++) {
+    const typedArray = new attribute.value.constructor(3);
+    const subarray = attribute.value.subarray(
+      (offset + i) * attribute.size,
+      (offset + i) * attribute.size + attribute.size
+    );
+    typedArray.set(subarray);
+    geometryVertices.push(new Vector3(typedArray));
```

Comment:
You are generating a lot of small objects here. It might be better to just copy the coordinates directly into the new Vector3 rather than messing with a temporary typed array.

This seems like a generic utility for converting flat arrays to nested arrays and could be part of `loaders.gl/math` module as first step and then move into math.gl.